mtp_api/apps/mtp_auth/serializers.py

```diff
@@ -225,25 +247,35 @@ def update(self, user, validated_data):
             updated_user.groups.clear()
 
             if make_user_admin:
-                updated_user.groups.add(user_admin_group)
+                self._make_user_admin(updated_user)
```

Comment:
I just spotted that this is causing a little bug when updating a 'Security' user to be a `UserAdmin`/FIU: It wouldn't add them to FIU group as expected.

This is a consequence of using `_make_user_admin()` _just after_ their groups are cleared above (`updated_user.groups.clear()`) as this helper method checks is user is part of `Security` group but at that point they'll belong to no groups.

```python
    def _make_user_admin(new_user):
        new_user.groups.add(Group.objects.get(name='UserAdmin'))
        # We add the FIU group to enforce current requirement that Security UserAdmin's are synonymous with FIU
        if new_user.groups.filter(name='Security').exists():
            new_user.groups.add(Group.objects.get(name='FIU'))
```

Move this call after the role's groups are added back below:

```python
            for group in role.groups:
                updated_user.groups.add(group)
```

should solve this issue...

...but I'll probably fix in a separate PR after this PR is merged.

Reply:
FYI: Fixed in this other PR: https://github.com/ministryofjustice/money-to-prisoners-api/pull/652